tests/e2e/spec/tonedear.e2e.spec.js

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable no-undef */
+import { By, until } from 'selenium-webdriver';
+import assert from 'assert';
+import paths from '../paths.js';
+
+const BROWSERSTACK = !!process.env.BROWSERSTACK_LOCAL_IDENTIFIER;
+const port = BROWSERSTACK ? '8099' : '8080';
+const tonedearBaseFile = BROWSERSTACK ? '/tests/zims/tonedear/tonedear.com_en_2024-09.zim' : paths.tonedearBaseFile;
+
+/**
+ * Run the tests
+ * @param {WebDriver} driver Selenium WebDriver object
+ * @param {Array} modes Array of modes to run the tests in
+ * @param {boolean} keepDriver Whether to keep the driver open after the tests have run
+ * @returns {Promise<void>} A Promise for the completion of the tests
+ */
+function runTests (driver, modes, keepDriver) {
+    let browserName, browserVersion;
+    driver.getCapabilities().then(function (caps) {
+        browserName = caps.get('browserName');
+        browserVersion = caps.get('browserVersion');
+        console.log('\nRunning Tonedear tests on: ' + browserName + ' ' + browserVersion);
+    });
+
+    // Set the implicit wait to 3 seconds
+    driver.manage().setTimeouts({ implicit: 3000 });
+
+    // Run in both jquery and serviceworker modes by default
+    if (!modes) {
+        modes = ['jquery', 'serviceworker'];
+    }
+
+    modes.forEach(function (mode) {
+        const serviceWorkerAPI = true;
+        describe('Tonedear test ' + (mode === 'jquery' ? '[JQuery mode]' : '[SW mode]'), function () {
+            this.timeout(60000);
+            this.slow(10000);
+
+            it('Load Kiwix JS and check title', async function () {
+                await driver.get('http://localhost:' + port + '/dist/www/index.html?noPrompts=true');
+                await driver.sleep(1300);
+                await driver.navigate().refresh();
+                await driver.sleep(800);
+                const title = await driver.getTitle();
+                assert.equal('Kiwix', title);
+            });
+
+            it('Switch to ' + mode + ' mode', async function () {
+                const modeSelector = await driver.wait(until.elementLocated(By.id(mode + 'ModeRadio')));
+                await driver.wait(async function () {
+                    const elementIsVisible = await driver.executeScript(
+                        'var el=arguments[0]; el.scrollIntoView(true); setTimeout(function () {el.click();}, 50); return el.offsetParent;',
+                        modeSelector
+                    );
+                    return elementIsVisible;
+                }, 5000);
+                await driver.sleep(1300);
+
+                if (mode === 'serviceworker' && !serviceWorkerAPI) {
```

Comment:
This code block will never run because serviceWorkerAPI is stuck on true. You need to test whether serviceWorker API is supported or not. See the Gutenberg test to see how it's done.